File: server/router/client/language.go
package client

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type LanguageRouter struct{}

// InitLanguageRouter 初始化 语言管理 路由信息
func (s *LanguageRouter) InitLanguageRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	languageRouter := Router.Group("language").Use(middleware.OperationRecord())
	languageRouterWithoutAuth := PublicRouter.Group("language")

	var languageApi = v1.ApiGroupApp.ClientApiGroup.LanguageApi
	{
		languageRouter.POST("createLanguage", languageApi.CreateLanguage)       // 创建语言
		languageRouter.DELETE("deleteLanguage", languageApi.DeleteLanguage)     // 删除语言
		languageRouter.PUT("updateLanguage", languageApi.UpdateLanguage)        // 更新语言
		languageRouter.GET("getLanguageList", languageApi.GetLanguageList)      // 获取语言列表（管理端）
	}
	{
		languageRouterWithoutAuth.GET("getEnabledLanguages", languageApi.GetEnabledLanguages) // 客户端获取启用的语言
	}
}

File: web/src/api/client/language.js
import service from '@/utils/request'

// @Tags Language
// @Summary 创建语言
export const createLanguage = (data) => {
  return service({
    url: '/language/createLanguage',
    method: 'post',
    data
  })
}

// @Tags Language
// @Summary 删除语言
export const deleteLanguage = (params) => {
  return service({
    url: '/language/deleteLanguage',
    method: 'delete',
    params
  })
}

// @Tags Language
// @Summary 更新语言
export const updateLanguage = (data) => {
  return service({
    url: '/language/updateLanguage',
    method: 'put',
    data
  })
}

// @Tags Language
// @Summary 获取语言列表（管理端）
export const getLanguageList = (params) => {
  return service({
    url: '/language/getLanguageList',
    method: 'get',
    params
  })
}

// @Tags Language
// @Summary 获取启用的语言列表（客户端）
export const getEnabledLanguages = () => {
  return service({
    url: '/language/getEnabledLanguages',
    method: 'get'
  })
}
